internal/exec/util/passwd.go

```diff
@@ -50,13 +50,25 @@ func appendIfStringSet(args []string, arg string, str *string) []string {
 }
 
 // EnsureUser ensures that the user exists as described. If the user does not
-// yet exist, they will be created, otherwise the existing user will be
-// modified.
+// yet exist, they will be created, otherwise the existing user will be modified.
+// If the `shouldExist` field is set to false and the user already exists, then
+// they will be deleted.
 func (u Util) EnsureUser(c types.PasswdUser) error {
 	exists, err := u.CheckIfUserExists(c)
 	if err != nil {
 		return err
 	}
+	if c.ShouldExist != nil && !*c.ShouldExist && exists {
+		args := []string{"--remove", "--root", u.DestDir, c.Name}
+		_, err := u.LogCmd(exec.Command(distro.UserdelCmd(), args...),
+			"deleting user %q", c.Name)
+		if err != nil {
+			return fmt.Errorf("failed to delete user %q: %v",
+				c.Name, err)
+		}
+		return nil
+	}
```

Comment:
I might be missing something but isn't there a case that we're not covering here (and in `EnsureGroup`)?

If a user is specified in the config with `shouldExist=false` and does NOT exist in the system then I think that we'll go past this block and start running `useradd`. `CheckIfUserExists` will return `false, nil` due to receiving `user.UnknownUserError` and there's nothing that handles immediately returning `nil` if `c.ShouldExist != nil && !c.ShouldExist && !exists`.

Reply:
Yeah, I missed that case. Let me go ahead and fix it.